core/services/pipeline/orm.go

```diff
@@ -221,36 +176,36 @@ func (o *orm) processNextUnfinishedRun(ctx context.Context, fn ProcessRunFunc) e
 	var pRun Run
 
 	err := postgres.GormTransaction(txContext, o.db, func(tx *gorm.DB) error {
-		err := tx.Raw(`
-		SELECT id FROM pipeline_runs
-		WHERE finished_at IS NULL
-		ORDER BY id ASC
-		FOR UPDATE SKIP LOCKED
-		LIMIT 1
-		`).Scan(&pRun).Error
-		if err != nil {
-			return errors.Wrap(err, "error finding next pipeline run")
-		}
-		// NOTE: We have to lock and load in two distinct queries to work
-		// around a bizarre bug in gormv1.
-		// Trying to lock and load in one hit _sometimes_ fails to preload
-		// associations for no discernable reason.
-		err = tx.
```

Comment:
Is this bug fixed in gormv2?

Reply:
yeah I was unable to reproduce it with gorm v2

Reply:
Hoorah!